clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -2399,4 +2400,39 @@ public KafkaProducer<T, T> newKafkaProducer() {
         }
     }
 
+    @Test
+    void testSerializerMustNoBeANull() {
```

Comment:
This seems to be the same as `testNoSerializerProvided()`

Reply:
Right, it's not exactly the same. If you really want to keep it, maybe we can just add this check to the existing test?

Reply:
Okay, I have moved the check to `testNoSerializerProvided`